src/rules/noInternalModuleRule.ts

```diff
@@ -24,12 +24,29 @@ export class Rule extends Lint.Rules.AbstractRule {
 
 class NoInternalModuleWalker extends Lint.RuleWalker {
     public visitModuleDeclaration(node: ts.ModuleDeclaration) {
-        if (Lint.isNodeFlagSet(node, ts.NodeFlags.Namespace)) {
-            // ok namespace
-        } else if (node.name.kind === ts.SyntaxKind.Identifier) {
-            // for external modules, node.name will be a LiteralExpression instead of Identifier
+        if (this.isInternalModuleDeclaration(node)) {
             this.addFailure(this.createFailure(node.getStart(), node.getWidth(), Rule.FAILURE_STRING));
         }
         super.visitModuleDeclaration(node);
     }
+
+    private isInternalModuleDeclaration(node: ts.ModuleDeclaration): boolean {
+        // for external modules, node.name will be a LiteralExpression instead of Identifier
+        return node.name.kind === ts.SyntaxKind.Identifier &&
+            !(this.isNamespaceDeclaration(node) || this.isNestedDeclaration(node));
+    }
+
+    private isNamespaceDeclaration(node: ts.ModuleDeclaration): boolean {
+        // the declaration may be nested, so recurse until the nesting chain breaks or reaches a namespace node
+        return node != null &&
+            this.isNestedDeclaration(node) &&
+            this.isNamespaceDeclaration(<ts.ModuleDeclaration> node.parent) ||
+            Lint.isNodeFlagSet(node, ts.NodeFlags.Namespace);
+    }
+
+    private isNestedDeclaration(node: ts.ModuleDeclaration): boolean {
```

Comment:
please add a comment above this method explaining that "nested" means something of the form "a.b.c"